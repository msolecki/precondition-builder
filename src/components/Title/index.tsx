import React from 'react'
import PropTypes from 'prop-types'
import {Typography} from '@material-ui/core'

interface TitleProps {
    text: string;
}

const Title: React.FC<TitleProps> = (props: TitleProps): React.ReactElement => {
    return (
        <Typography variant="h5" component="h5">
            {props.text}
        </Typography>
    )
}

Title.propTypes = {
    text: PropTypes.string.isRequired,
}

Title.defaultProps = {
    text: '',
}

export default Title
