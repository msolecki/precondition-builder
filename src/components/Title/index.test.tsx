import * as React from 'react'
import {create, ReactTestRenderer} from 'react-test-renderer'

import Title from './index'


let snapshot: ReactTestRenderer

beforeEach(() => {
    const title = <Title text='test title'/>


    snapshot = create(title)
})

describe('<Title />', () => {
    test('it matches the snapshot', () => {
        expect(snapshot.toJSON()).toMatchSnapshot()
    })
})
