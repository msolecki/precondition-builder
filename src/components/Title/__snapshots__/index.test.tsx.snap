// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> it matches the snapshot 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5"
>
  test title
</h5>
`;
